### PR TITLE
Add basic support for button entity

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

https://github.com/home-assistant/core/pull/57642

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/entity-rows/hui-button-entity-row.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import "@material/mwc-button/mwc-button";
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/hui-button-entity-row.ts:1]: What is the difference in the special button row and this? Could we combine them or is there a reason for two

Comment (bramkragten) [reply src/panels/lovelace/entity-rows/hui-button-entity-row.ts:1]: One is based on an entity, and the other is just config

File: src/panels/lovelace/entity-rows/hui-button-entity-row.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row .hass=${this.hass} .config=${this._config}>
+        <mwc-button
+          @click=${this._pressButton}
+          .disabled=${stateObj.state === UNAVAILABLE}
+        >
+          ${this.hass.localize("ui.card.button.press")}
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-button-entity-row.ts:56]: I think that this row can be like this:
- name: Device name
- button label: entity name

Comment (bramkragten) [reply src/panels/lovelace/entity-rows/hui-button-entity-row.ts:56]: And if the entity doesn't belong to a device?

Comment (balloob) [reply src/panels/lovelace/entity-rows/hui-button-entity-row.ts:56]: Then we use "press" ? 🤷 

But right, it is weird…

Comment (frenck) [reply src/panels/lovelace/entity-rows/hui-button-entity-row.ts:56]: I think we should use press now... we could use device classes to influence this maybe?
For example, a DEVICE_CLASS_RESTART for this one, to show "RESTART" and ask for confirmation for the future?

Comment (balloob) [reply src/panels/lovelace/entity-rows/hui-button-entity-row.ts:56]: ok.

**Review by balloob — APPROVED** [review]

**Pull request merged**